### PR TITLE
release 1.0

### PR DESCRIPTION
**Pull request by Toru8P**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/codec/openapi/utils/JsonUtils.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+fun JsonNode.getRequiredArray(fieldName: String, required: Boolean) : ArrayNode? {
+    val field = get(fieldName)
+    if (required && field==null) {
+        error {"$fieldName array field was required!"}
+    }
+    if (field!=null && !field.isArray) {
+        error {"$fieldName field of json isn't array!"}
+    }
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/openapi/utils/JsonUtils.kt:77]:
```suggestion
    val field = getRequiredField(fieldName, required)

    if(field != null && !field.isArray) {
        error("$fieldName field of json isn't array!")
    }

```

File: src/main/kotlin/com/exactpro/th2/codec/openapi/utils/JsonUtils.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    return field as? ArrayNode
+}
+
+fun JsonNode.validateAsBoolean() : Boolean {
+    return if (isBoolean) {
+        asText().toBoolean()
+    } else error("Cannot convert $this to Boolean")
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/openapi/utils/JsonUtils.kt:84]:
```suggestion
fun JsonNode.validateAsBoolean() : Boolean = when {
    isBoolean -> asBoolean()
    else -> error("Cannot convert $this to Boolean")
```

File: src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        "delete" -> {
+            return delete
+        }
+        "post" -> {
+            return post
+        }
+        else -> {
+            error("Unsupported method inside message")
+        }
+    }
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt:42]:
```suggestion
private fun PathItem.getByMethod(method: String) = when (method.lowercase()) {
    "get" -> get
    "put" -> put
    "delete" -> delete
    "post" -> post
    else -> error("Unsupported method inside message")
```

File: src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+@Suppress("UNCHECKED_CAST")
+fun PathItem.getMethods() : Map<String, Operation> {
+    return listOf("get", "put", "delete", "post").map {
+        it to getByMethod(it)
+    }.filter { it.second != null }.toMap() as Map<String, Operation>
+}
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt:50]: Can be replaced with a constant map

File: src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+enum class JsonSchemaTypes(val type: String) {
+    ARRAY("array"), OBJECT("object");
+
+    companion object {
+        fun getType(type: String) : JsonSchemaTypes? {
+            values().forEach {
+                if (it.type == type) return it
+            }
+            return null
+        }
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/openapi/utils/OpenApiUtils.kt:76]:
```suggestion
        fun getType(type: String) : JsonSchemaTypes? = values().first { it.type == type }
```

**Review by cordwelt — COMMENTED** [review]
Very superficial code-review

**Pull request merged**